core/src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -50,7 +50,7 @@
     /**
      * Sets the arbitrary name to be assigned to the query (see named queries).
      */
-    QB queryName(String queryName);
+    QueryBuilder queryName(String queryName);
```

Comment:
It works fine in practice because query builders are always constructed with their concrete type. With this pull request already, none of the client code that we had in the test folders had to change.

I will add the note to the javadocs of QueryBuilder (pointing out that extending AbstractQueryBuilder does it for you).